pkg/proxy/ipvs/proxier.go

```diff
@@ -899,6 +899,23 @@ func (proxier *Proxier) OnEndpointSlicesSynced() {
 // EntryInvalidErr indicates if an ipset entry is invalid or not
 const EntryInvalidErr = "error adding entry %s to ipset %s"
 
+func getLocalAddrs() (map[string]bool, error) {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return nil, err
+	}
+
+ 	localAddrs := map[string]bool{}
+	for i := range addrs {
+		intf, _, err := net.ParseCIDR(addrs[i].String())
+		if err != nil {
+			return nil, err
+		}
+		localAddrs[intf.String()] = true
```

Comment:
Does this end up being the same equality check as `utilproxy.IsLocalIP`?

In `utilproxy.IsLocalIP` we have:
```
if net.ParseIP(ip).Equal(intf) {
    return true, nil
}
```

We're no longer doing the `ip.Equal` check and assuming the string values are the same.

Reply:
I also see that `Equal` from `net.IP` considers an IPv4 address in IPv6 form equal to it's IPv4 form. 

```
// Equal reports whether ip and x are the same IP address.
// An IPv4 address and that same address in IPv6 form are
// considered to be equal.
func (ip IP) Equal(x IP) bool {
	if len(ip) == len(x) {
		return bytealg.Equal(ip, x)
	}
	if len(ip) == IPv4len && len(x) == IPv6len {
		return bytealg.Equal(x[0:12], v4InV6Prefix) && bytealg.Equal(ip, x[12:])
	}
	if len(ip) == IPv6len && len(x) == IPv4len {
		return bytealg.Equal(ip[0:12], v4InV6Prefix) && bytealg.Equal(ip[12:], x)
	}
	return false
}
```

Reply:
yes, net.IP.Equal seems more safe for ipv6, I will rollback this compare logic.